qiskit/circuit/quantumcircuit.py

```diff
@@ -2043,6 +2043,9 @@ def _assign_parameter(self, parameter, value):
         if isinstance(value, ParameterExpression):
             entry = self._parameter_table.pop(parameter)
             for new_parameter in value.parameters:
+                if new_parameter in self._parameter_table:
+                    entry.extend(self._parameter_table[new_parameter])
```

Comment:
Instead of re-assigning the value of entry, would it be more efficient to just update the list inplace? Like
```python
if new_parameter in self._parameter_table:
    self._parameter_table[new_parameter].extend(entry)
else:
    self._parameter_table[new_parameter] = entry
```
?

Reply:
Yes indeed. I updated accordingly. Thanks for the suggestion @Cryoris .